observability/tracing.go

```diff
@@ -117,3 +134,72 @@ func StartSpan(ctx context.Context, funcName string, attrs ...attribute.KeyValue
 
 	return ctx, func() {}
 }
+
+// getExporter returns the exporter based on the configuration options
+// Jaeger collector, OpenTelemetry using http or grpc
+func getExporter(collectorURL string) (sdktrace.SpanExporter, error) {
+	var exporter sdktrace.SpanExporter
+	var err error
+
+	tracingOpt := config.Get().Server.Observability.Tracing
+
+	// Jaeger collector
+	if tracingOpt.CollectorType == JAEGER {
+		log.Debugf("Creating Jaeger collector with URL %s", collectorURL)
+		exporter, err = jaeger.New(jaeger.WithCollectorEndpoint(jaeger.WithEndpoint(collectorURL)))
+		return exporter, err
+	} else {
+		if tracingOpt.CollectorType == OTEL {
+			// OpenTelemetry collector
+			if tracingOpt.Otel.Protocol == HTTP || tracingOpt.Otel.Protocol == HTTPS {
+				tracingOptions := otlptracehttp.WithRetry(otlptracehttp.RetryConfig{
+					Enabled:         true,
+					InitialInterval: 1 * time.Nanosecond,
+					MaxInterval:     1 * time.Nanosecond,
+					// Never stop retry of retry-able status.
+					MaxElapsedTime: 0,
+				})
+				var client otlptrace.Client
+
+				if tracingOpt.Otel.Protocol == HTTP {
+					log.Debugf("Creating OpenTelemetry collector with URL http://%s", collectorURL)
+
+					client = otlptracehttp.NewClient(otlptracehttp.WithEndpoint(collectorURL),
+						otlptracehttp.WithInsecure(),
+						tracingOptions,
+					)
+				} else {
+					log.Debugf("Creating OpenTelemetry collector with URL https://%s", collectorURL)
+
+					client = otlptracehttp.NewClient(otlptracehttp.WithEndpoint(collectorURL),
+						tracingOptions,
+					)
+				}
+
+				ctx := context.Background()
+				exporter, err := otlptrace.New(ctx, client)
+				return exporter, err
+			} else {
+				if tracingOpt.Otel.Protocol == GRPC {
+					log.Debugf("Creating OpenTelemetry grpc collector with URL %s", collectorURL)
+					ctx := context.Background()
+					ctx, cancel := context.WithTimeout(ctx, time.Second)
+					defer cancel()
+
+					// TODO: Support TLS
+					exporter, err := otlptracegrpc.New(ctx, otlptracegrpc.WithInsecure(),
+						otlptracegrpc.WithEndpoint(collectorURL),
+						otlptracegrpc.WithDialOption(grpc.WithBlock()),
+					)
+
+					return exporter, err
+				} else {
+					return nil, errors.New("Error in configuration options getting the observability exporter")
```

Comment:
Actually, make these error messages more specified. This error and the one on line 201 are identical - when you see this, you have no idea if its the protocol that is bad or the collector type that is bad.
```suggestion
					return nil, errors.New("Error in configuration options getting the observability exporter - invalid otel.protocol.")
```